### PR TITLE
Initial MVP version

### PR DESCRIPTION
**Pull request by schadalawada**
Onboarding terraform module to create network edge NGINX device

### CODE REVIEW

File: README.md
Comment (ocobleseqx) [inline README.md:1]: I think this title is for another project

File: README.md
Comment (ocobleseqx) [inline README.md:6]: It seems that Google provider reference shouldn't be here

File: examples/nginx-ha/main.tf
Comment (ocobleseqx) [inline examples/nginx-ha/main.tf:34]: In your README.md and examples, avoid using any credentials that look like real credentials. These credentials will be flagged and reviewed by the security team.

In the README files in each example, the defined steps are assuming there is nothing to modify to make it work: clone -> init -> apply, so if there's something that must be modified like that public_key you should better define a variable or use the terraform tls provider to generate a new one in the example. On the other hand, you can reduce the variables in the examples, like `metro_code_primary`, you can take any one like "SV" so users can get an idea of the expected format for that field

File: versions.tf
Comment (ocobleseqx) [inline versions.tf:9]: Notice the version constraint "~>" to allow new patch releases within a specific minor release. 
This is not mandatory but, we are including `provider_meta` in all of our modules so that we can update the user agent accordingly, it allows us to keep track of module usage.

More info https://github.com/equinix-labs/equinix-labs/blob/main/terraform-module-standards.md#equinix-provider

```suggestion
terraform {
  required_version = ">= 0.13"
  required_providers {
    equinix = {
      source  = "equinix/equinix"
      version = "~> 1.14"
    }
  }
  provider_meta "equinix" {
    module_name = "terraform-equinix-network-edge-nginx"
  }
}
```

**Review by ocobleseqx — CHANGES_REQUESTED** [review]
I added some comments that have to do with some Equinix standards, but not with functionality. It looks good but I need to test it, but in the meantime could you address my comments?

File: README.md
Comment (displague) [inline README.md:73]: @schadalawada The requirements, variables, and outputs of this module can be automatically maintained by workflows that come included in `.github`. You'll need to keep the `<!-- BEGIN_TF_DOCS -->` section to keep that behavior (recommended).

File: examples/nginx-ha/main.tf
Comment (displague) [reply examples/nginx-ha/main.tf:34]: @schadalawada if you don't use the Terraform resource to generate the public key, we should use a variable here so that this example is usable without Terraform file modification.  We could then use CI to test this example by defining environment variables to override the SSH key (i.e. `TF_VAR_user_ssh_key`).

File: CODEOWNERS
Comment (ocobleseqx) [reply CODEOWNERS]: @schadalawada please add the CODEOWNERS file

```
# TEMPLATE: add your username after terraform
# TEMPLATE: * equinix-labs/terraform myusername
* @equinix-labs/terraform
```

File: README.md
Comment (ocobleseqx) [reply README.md:73]: @schadalawada I saw you added the comment back. We have a GH action to generate documentation that uses that block. It will generate next sections for you: Requirements, Providers, Modules, Resources, Inputs, Outputs. 

Please remove all those sections in the README file and just add this code where you want them

```
<!-- BEGIN_TF_DOCS -->
<!-- END_TF_DOCS -->
```

**Comment by ocobleseqx** [comment]
I approved it but I realized a couple of things, which I added in the open conversations, once addressed we can merge it

**Review by poornimayampati — APPROVED** [review]

**Review by displague — APPROVED** [review]

File: README.md
Comment (displague) [reply README.md:73]: @schadalawada added BEGIN_TF_DOCS blocks. These are not in all the correct places, some custom content will be overridden by CI and some of the custom content blocks (in examples) is missing BEGIN_TF_DOCS. I'm going to resolve this thread so we can merge this. We'll want to chase this with corrections to the location of BEGIN_TF_DOCS blocks.

**Pull request merged**
